lib/oli/delivery/hierarchy.ex

```diff
@@ -265,6 +265,36 @@ defmodule Oli.Delivery.Hierarchy do
     end)
   end
 
+  @doc """
+  Build a map that contains a list of ancestor resource_ids for each node
```

Comment:
It would be helpful to include more documentation here, especially indicating that a gated resource list entry will contain itself.  maybe icluding a sample hierarchy and sample set of gates, with a resultant gating index is a good idea.  We want to be able to look at this code two years from now and immediately undertand what it is building and why